tests/test_version_traversal.py

```diff
@@ -1176,6 +1176,59 @@ def test_dependencies_are_normalized(self):
         )
 
 
+class DependsOnBranchTestFive(MigrationTest):
+    @classmethod
+    def setup_class(cls):
+        """
+        issue #1373
+
+        Structure::
+
+            <base> -> a1 ------+
+                      ^        |
+                      |        +-> bmerge
+                      |        |
+                      +-- b1 --+
+        """
+        cls.env = env = staging_env()
+        cls.a1 = env.generate_revision("a1", "->a1")
+        cls.b1 = env.generate_revision(
+            "b1", "->b1", head="base", depends_on="a1"
+        )
+        cls.bmerge = env.generate_revision(
+            "bmerge", "bmerge", head=[cls.a1.revision, cls.b1.revision]
+        )
+
+    @classmethod
+    def teardown_class(cls):
+        clear_staging_env()
+
+    def test_downgrade_to_depends_on(self):
+        # Upgrade from a1 to b1 just has heads={"b1"}.
+        self._assert_upgrade(
+            self.b1.revision,
+            self.a1.revision,
+            [self.up_(self.b1)],
+            {self.b1.revision},
+        )
+
+        # Upgrade from b1 to bmerge just has {"bmerge"}
+        self._assert_upgrade(
+            self.bmerge.revision,
+            self.b1.revision,
+            [self.up_(self.bmerge)],
+            {self.bmerge.revision},
+        )
+
+        # Downgrading from bmerge to a1 should return back to heads={"b1"}.
+        self._assert_downgrade(
+            self.a1.revision,
+            self.bmerge.revision,
+            [self.down_(self.bmerge)],
+            {self.b1.revision},
```

Comment:
**Federico Caselli** (CaselIT) wrote:

this kinda seems an error, since I would expect that going from bmerge to a1 would do `bmerge -> b1 -> a1`. But this is a different issue so let's leave it as is

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/alembic/+/5027

Reply:
I think the current behavior makes sense. b1 does not have a1 as a _down revision_, it has it as a _depends_on_. Downgrading b1 would be unexpected;

Reply:
Well but now it's even stranger: you ask to downgrade to a1 but it downgrades to b1 instead.

Still that's not something for really connected to this issue, just noticed while looking at the test